include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -127,8 +126,7 @@ void hipLaunchKernelGGLImpl(
     hipStream_t stream,
     void** kernarg) {
 
-    const auto& kd = hip_impl::get_program_state().kernel_descriptor(function_address, 
-                                                               target_agent(stream));
+    const auto& kd = hip_impl::get_kernel_by_address(function_address, stream);
```

Comment:
We need to call get_program_state here so that it gets inlined such that it will get the module specific copy of the program_state static object.   Moving the call to get_program_state() into functional_grid_launch.inl would break some programs.

Reply:
I have changed it a little bit, to keep `hip_get_kernel_by_address` stateless for any static program_state.